### PR TITLE
Promote release_channel to GA for GKE

### PR DESCRIPTION
**Pull request by upodroid**
Fixes: https://github.com/terraform-providers/terraform-provider-google/issues/6390

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: added `release_channel` to `google_container-cluster` (GA)
```

### CODE REVIEW

**Comment by ndmckinley** [comment]
I'm suspicious that this change broke the GKE tests.

https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloud_ProviderGoogleCloudGoogleProject/137992

File: third_party/terraform/resources/resource_container_cluster.go.erb [hunk 1]
```diff
@@ -2939,7 +2939,6 @@ func expandDatabaseEncryption(configured interface{}) *containerBeta.DatabaseEnc
 	}
 }
 
-<% unless version == 'ga' -%>
 func expandReleaseChannel(configured interface{}) *containerBeta.ReleaseChannel {
```
Comment (ndmckinley) [inline third_party/terraform/resources/resource_container_cluster.go.erb:2942]: I think you need a nil check here before the attempt at a cast; if l == nil { return nil }.

**Comment by rileykarson** [comment]
Oh whoops, I forgot that VCR doesn't run on community PRs. Makes sense! I see you sent https://github.com/GoogleCloudPlatform/magic-modules/pull/3836.

**Comment by upodroid** [comment]
Hi, was it broken in the beta one too?

**Comment by ndmckinley** [comment]
It didn't seem to be, which is a little odd!  I am waiting for the tests to come back to be sure that this is the issue.

**Pull request merged**